atsign-foundation/at_server
# feat: allow update of enrollment access for approved enrollments

**- closes https://github.com/atsign-foundation/at_server/issues/1499**

**- What I did**
- allow update of enrollment access of approved enrollments for specific namespaces
- refactor enroll verb handler
- replace exceptions with appropriate error codes and error messages
- Introduce new class EnrollmentResponse to handle responses and data in EnrollVerbHandler
- add enrollment status of each enrollment to enroll:list response

**- How I did it**
- Introduce a new apkam operation called update
- an update request needs to be placed for an already existing enrollments with updated namespaces
- an apkam authenticated connection needs to approve that request

**- How to verify it**
- new unit and functional tests added to verify enrollment access update behaviour
- existing tests modified to match refactoring

**- Description for the changelog**

feat: allow update of enrollment access for approved enrollments
